workers/unity/Packages/io.improbable.gdk.debug/WorkerInspector/EntityList.cs

```diff
@@ -0,0 +1,268 @@
+
+                            if (chunk.Has(metadataComponentType))
+                            {
+                                metadataArray = chunk.GetNativeArray(metadataComponentType);
+                            }
+
+                            var entityIdArray = chunk.GetNativeArray(spatialOSComponentType);
+                            var entities = chunk.GetNativeArray(ecsEntityType);
+
+                            for (var i = 0; i < entities.Length; i++)
+                            {
+                                var data = new EntityData(entityIdArray[i].EntityId, metadataArray?[i].EntityType);
+                                underlying.Add(data);
+                            }
+                        }
+                    }
+                }
+
+                underlying.Sort();
+
+                Calculate();
+            }
+        }
+
+        private readonly struct EntityData : IComparable<EntityData>, IComparable, IEquatable<EntityData>
+        {
+            public readonly EntityId EntityId;
+            public readonly string Metadata;
+
+            public EntityData(EntityId entityId, string metadata)
+            {
+                EntityId = entityId;
+                Metadata = metadata;
+            }
+
+            public bool Matches(SearchParameters searchParameters)
+            {
+                if (searchParameters.EntityId.HasValue)
+                {
+                    return searchParameters.EntityId.Value == EntityId;
+                }
+
+                if (!string.IsNullOrEmpty(searchParameters.SearchFragment))
+                {
+                    return Metadata.ToLower().Contains(searchParameters.SearchFragment);
+                }
+
+                return true;
+            }
+
+            public override string ToString()
+            {
+                return Metadata != null ? $"{Metadata} ({EntityId})" : EntityId.ToString();
+            }
+
+            public int CompareTo(EntityData other)
+            {
+                return EntityId.CompareTo(other.EntityId);
+            }
+
+            public int CompareTo(object obj)
+            {
+                if (ReferenceEquals(null, obj))
+                {
+                    return 1;
+                }
+
+                return obj is EntityData other ? CompareTo(other) : throw new ArgumentException($"Object must be of type {nameof(EntityData)}");
+            }
+
+            public bool Equals(EntityData other)
+            {
+                return EntityId.Equals(other.EntityId) && Metadata == other.Metadata;
+            }
+
+            public override bool Equals(object obj)
+            {
+                return obj is EntityData other && Equals(other);
+            }
+
+            public override int GetHashCode()
+            {
+                unchecked
+                {
+                    return (EntityId.GetHashCode() * 397) ^ (Metadata != null ? Metadata.GetHashCode() : 0);
```

Comment:
Is 397 a magic number? I'm not familiar with what it means here.

Reply:
This is the magic auto-impl of `GetHashCode()` by Rider. I believe 397 is a prime number which plays some role in this.